test/functional/test_framework/util.py

```diff
@@ -385,6 +385,17 @@ def set_node_times(nodes, t):
     for node in nodes:
         node.setmocktime(t)
 
+def disconnect_nodes(from_connection, node_num):
+    timeout = 5
+    while True:
+        peer_ids = [peer['id'] for peer in from_connection.getpeerinfo() if "testnode%d" % node_num in peer['subver']]
+        if not peer_ids:
+            break
+        [from_connection.disconnectnode(nodeid=peer_id) for peer_id in peer_ids]
```

Comment:
This happens every time through the loop. Is that your intention? Doesn't this only have to be done once?